### DISCUSSION TITLE
tough-cookie 5

### DISCUSSION THREAD

**Discussion by awaterma** [salesforce/tough-cookie#255] [Ideas]
I wanted to open up some discussions about tough-cookie 5. We are looking at some major changes around testing and typescript support. I think we'd also like to consider turning off `special use domains` by default; although I worry about what effects that might have downstream (please see issue #246).

**Comment by colincasey** [comment]
Here are a couple of ideas that I think should be considered:

* **TypeScript support**
   *This would be an ergonomic improvement for developers & may help to reduce implementation errors for maintainers*

* **Migrate Test Harness from [Vows](https://github.com/vowsjs/vows) to [Jest](https://jestjs.io/)**
   *This would be a tooling improvement for maintainers and potential contributors*

* **Refactor the codebase so that the Cookie Specification and what is implemented can be easily compared**
   *Another internal improvement that I think will improve maintainability, understanding, and discussions around spec compliance*

Most of the suggestions above can be done with little to no change to the public-facing API but it would be nice to change the default configurations to enable the strictest interpretation of the spec.

**Comment by colincasey** [comment]
we should also consider alternatives to dependencies we have such as `psl` versus `tldts` since we have had performance tickets like #239 raised

**Comment by graffhyrum** [comment]
Hi all, 
I'm checking in to see if there's any roadmap or estimate for the v5 release, or the psl replacement. I hunted around in the open and closed issues and eventually came across #346, but I hadn't seen any other hints about when it would make it into a release. 
Thanks for all the work on this!

**Reply by wjhsf** [reply]
We don't have anything official, but we're probably going to release v5 in the next month or so. I think we've done just about everything we care to include in the release, so we'll hopefully be scheduling it soon!

**Comment by awaterma** [comment]
Closing this, as we released version 5 today. If you have any comments for discussion; please post on our announcement for that release!
